packages/rpc-types/src/rpc-api.ts

```diff
@@ -12,6 +13,17 @@ type RpcTransportConfig = Readonly<{
 export interface IRpcTransport {
     <TResponse>(config: RpcTransportConfig): Promise<TResponse>;
 }
+export type IRpcTransportDevnet = IRpcTransport & { '~cluster': 'devnet' };
+export type IRpcTransportTestnet = IRpcTransport & { '~cluster': 'testnet' };
+export type IRpcTransportMainnet = IRpcTransport & { '~cluster': 'mainnet' };
+export type IIRpcTransport = IRpcTransport | IRpcTransportDevnet | IRpcTransportTestnet | IRpcTransportMainnet;
```

Comment:
`IICaptain` 🫡

This double `I` feels very strange to me. 😅

We may not even need this `IIRpcTransport` type since `IRpcTransportDevnet`, for instance, already extends `IRpcTransport`.

Also, I'm still making my way up the stack but I still feel like these types could be part of `@solana/rpc-transports`.

Reply:
> We may not even need this IIRpcTransport type since IRpcTransportDevnet, for instance, already extends IRpcTransport.

Yeah I thought about this, too, but to me it felt easier to read/follow the type conditionals with these explicit.

Reply:
How bout `IRpcTransportWithCluster`?

Reply:
But it's a union that includes one transport without a cluster, so that could be a bit confusing. Unless I use `IRpcTransport | IRpcTransportWithCluster` everywhere?

Reply:
`IRpcTransport | IRpcTransportWithCluster` Yes, where applicable.

Personally though, I'd remove `IIRpcTransport` altogether. I can't see a valid reason to keep it by looking at this PR only (but maybe I'm missing some stack context).